internal/cliwrapper/cliwrapper.go

```diff
@@ -15,12 +15,18 @@ import (
 type cliWrapper struct {
 	podmanFullPath string
 	logger         log.Logger
+	runtimeContext []string
 }
 
-func NewCliWrapper(fullPath string, logger log.Logger) CliWrapper {
+func NewCliWrapper(fullPath string, logger log.Logger, context string) CliWrapper {
+	// Specify podman --connection string if provided
+	if context != "" {
+		context = "--connection=" + context
+	}
 	return &cliWrapper{
 		podmanFullPath: fullPath,
 		logger:         logger,
+		runtimeContext: []string{context},
 	}
 }
```

Comment:
This produces an awkward result when `context` is an empty string:  it adds an empty argument to the Podman command line which produces weird effects (like making the relevant tests fail!  🙃).

I've [coded around that](https://github.com/ewchong/arcaflow-engine-deployer-podman/pull/1) (by not adding the context when it would contribute an empty argument), but it's kind of ugly, and it would be better if there were some other approach (like leaving the `runtimeContext` slice _empty_, instead of putting an empty string into it).

Reply:
Continued [below](https://github.com/arcalot/arcaflow-engine-deployer-podman/pull/61#discussion_r1583655416).